XenModel/XenAPI-Extensions/VMPP.cs

```diff
@@ -166,6 +167,95 @@ public override string Description
             get { return name_description; }
         }
 
+        public bool is_enabled
+        {
+            get { return this.is_policy_enabled; }
+        }
+
+        public bool is_running
+        {
+            get { return this.is_backup_running; }
+        }
+
+        public bool is_archiving
+        {
+            get { return this.is_archive_running; }
+        }
+
+        public Type _Type
+        {
+            get {return typeof(VMPP);}
+        }
+
+        public List<PolicyAlert> PolicyAlerts
+        {
+            get { return RecentAlerts;}
+        }
+
+        public bool hasArchive
+        {
+            get { return true; }
+        }
+        
+        public void set_vm_policy(Session session, string _vm, string _value)
+        {
+            VM.set_protection_policy(session, _vm, _value);
+        }
+
+        public void do_destroy(Session session, string _policy)
+        {
+            VMPP.destroy(session, _policy);
+        }
+
+        public string run_now(Session session, string _policy)
+        {
+            return VMPP.protect_now(session, _policy);
+        }
+
+        public void set_is_enabled(Session session, string _policy, bool _is_enabled)
+        {
+            VMPP.set_is_policy_enabled(session, _policy, _is_enabled);
+        }
+
+        public PureAsyncAction getAlertsAction(IVMPolicy policy, int hoursfromnow)
+        {
+            return new GetVMPPAlertsAction((VMPP)policy, hoursfromnow);
+        }
+
+        public policy_frequency policy_frequency
+        {
+            get { return ((policy_frequency)(backup_frequency)); }
+            set { backup_frequency = (vmpp_backup_frequency)value; }
+        }
+
+        public Dictionary<string, string> policy_schedule 
+        {
+            get { return backup_schedule; }
+            set { backup_schedule = value; }
+        }
+
+        public long policy_retention 
+        {
+            get { return backup_retention_value; }
+            set { backup_retention_value = value; }
+        }
+
+        public policy_backup_type policy_type 
+        {
+            get { return (policy_backup_type)backup_type; }

```

Comment:
This type of code is not really safe. It only happens to work because the policy_backup_type enum is in the same order as the vmpp_backup_type enum, with the additional items not present in vmpp_backup_type at the end. But you shouldn't make such assumptions about enums in case they get extended in a different way later.

I realise it's longer code, but it would be more idiomatic not to convert between different types of enums and to use a switch statement to enumerate through the backup types and return the right policy_backup type for each one (and vice versa in the setter).